src/backend/renderer/utils.rs

```diff
@@ -325,10 +330,14 @@ pub fn on_commit_buffer_handler(surface: &WlSurface) {
     }
 }
 
+/// Defines a view into the surface
 #[derive(Debug, Default, PartialEq, Clone, Copy)]
-pub(crate) struct SurfaceView {
+pub struct SurfaceView {
+    /// The logical source used for cropping
     pub src: Rectangle<f64, Logical>,
+    /// The logical destination size used for scaling
     pub dst: Size<i32, Logical>,
+    /// The logical offset for a sub-surface
```

Comment:
Hard to tell at this point but maybe this should become pub(crate) and then add getters() for these fields.